scripts/integration/docker-compose.datadog-agent.yml

```diff
@@ -6,14 +6,29 @@ services:
     network_mode: host
     environment:
       - DD_API_KEY=${CI_TEST_DATADOG_API_KEY}
+      - DD_APM_ENABLED=false
       - DD_LOGS_ENABLED=true
       - DD_LOGS_CONFIG_LOGS_DD_URL=0.0.0.0:8080
       - DD_LOGS_CONFIG_LOGS_NO_SSL=true
       - DD_LOGS_CONFIG_USE_HTTP=true
       - DD_HEALTH_PORT=8182
+      - DD_CMD_PORT=5001
+      - DD_USE_DOGSTATSD=false
     volumes:
       - ${PWD}/tests/data/datadog-agent/conf.yaml:/etc/datadog-agent/conf.d/test.d/conf.yaml
 
+  datadog-trace-agent:
+    # Using 7.31.0 image to have the ability mimic tracing lib using json
+    image: datadog/agent:7.31.0
```

Comment:
Will this also work with newer agents?

Reply:
Integration tests uses agent 7.31 because it supports ingesting json traces from tracer lib, making it easy to mock a trace and send that to a real trace agent to generate a real-life trace.

Regarding the trace-agent<->datadog trafic, the trace format has been recently updated (https://github.com/DataDog/datadog-agent/pull/9757), thus Vector will need to be updated subsequently to also support the newer format.

Reply:
Ah, gotcha, is that follow up work represented somewhere?